shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -88,62 +138,80 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             Append append = (Append) msg;
             Session session = setupSegments.get(new SimpleImmutableEntry<>(append.segment, append.getWriterId()));
             validateAppend(append, session);
-            if (!append.segment.equals(segmentBeingAppendedTo) || !append.getWriterId().equals(writerIdPerformingAppends)) {
-                breakFromAppend(null, null, out);
-            }
             final ByteBuf data = append.getData().slice();
             final int msgSize = data.readableBytes();
             final AppendBatchSizeTracker blockSizeSupplier = (appendTracker == null) ? null :
                     appendTracker.apply(append.getFlowId());
 
-            session.lastEventNumber = append.getEventNumber();
-            session.eventCount++;
             if (blockSizeSupplier != null) {
                 blockSizeSupplier.recordAppend(append.getEventNumber(), msgSize);
             }
-            if (bytesLeftInBlock == 0) {
-                currentBlockSize = msgSize + TYPE_PLUS_LENGTH_SIZE;
-                if (blockSizeSupplier != null) {
-                    currentBlockSize = Math.max(currentBlockSize, blockSizeSupplier.getAppendBlockSize());
+            if ((bytesLeftInBlock == 0) ||
```

Comment:
This method is getting rather long. Would it make it easier to test it if we factored out several smaller helper methods?

Reply:
yes.. I have done the refactoring now .. please have look